evap/staff/importers/json.py

```diff
@@ -0,0 +1,445 @@
+        course_type = self._get_course_type(data["type"]) if course_type is None else course_type
+        responsibles = self._get_user_profiles(data["lecturers"])
+        responsibles = self._remove_non_responsible_users(responsibles)
+        responsibles = self._get_users_with_longest_title(responsibles)
+        if not data["title_en"]:
+            data["title_en"] = data["title"]
+        course, created, changes = update_or_create_with_changes(
+            Course,
+            semester=self.semester,
+            cms_id=data["gguid"],
+            defaults={"name_de": data["title"], "name_en": data["title_en"], "type": course_type},
+        )
+        course.responsibles.set(responsibles)
+
+        if created:
+            self.statistics.new_courses.append(course)
+        elif changes:
+            self.statistics.updated_courses.append(course)
+
+        self.courses_by_gguid[data["gguid"]] = course
+
+        return course
+
+    def _import_course_programs(self, course: Course, data: ImportEvent) -> None:
+        programs = [
+            self._get_program(c["cprid"]) for c in data["courses"] if c["cprid"] not in settings.IGNORE_PROGRAMS
+        ]
+        course.programs.add(*programs)
+
+    def _import_course_from_unused_exam(self, data: ImportEvent) -> Course | None:
+        split_title = data["title"].split(":", 1)
+        if len(split_title) < 2:
+            return None
+        prefix = split_title[0].strip()
+
+        try:
+            course_type = CourseType.objects.get(import_names__contains=[prefix])
+        except CourseType.DoesNotExist:
+            return None
+
+        data["title"] = split_title[1].strip()
+        data["title_en"] = data["title_en"].split(":", 1)[1].strip() if ":" in data["title_en"] else data["title_en"]
+
+        return self._import_course(data, course_type)
+
+    # pylint: disable=too-many-locals
+    def _import_evaluation(self, course: Course, data: ImportEvent) -> Evaluation:
+        if "appointments" not in data:
+            self.statistics.warnings.append(
+                WarningMessage(obj=course.name, message="No dates defined, using default end date")
+            )
+            course_end = datetime.combine(self.default_course_end, time())
+        else:
+            last_appointment = max(data["appointments"], key=lambda x: x["end"])
+            course_end = datetime.strptime(last_appointment["end"], self.DATETIME_FORMAT)
+
+        if data["isexam"]:
+            # Set evaluation time frame of three days for exam evaluations:
+            evaluation_start_datetime = course_end.replace(hour=8, minute=0, second=0, microsecond=0) + timedelta(
+                days=1
+            )
+            evaluation_end_date = (course_end + timedelta(days=3)).date()
+
+            name_de = data["title"].split(" - ")[-1] if " - " in data["title"] else "Prüfung"
+            name_en = data["title_en"].split(" - ")[-1] if " - " in data["title_en"] else "Exam"
```

Comment:
Not a blocker; these translations should be handled through our po file, not here in code